### PR TITLE
Block-conversion

### PR DESCRIPTION
**Pull request by Rahul-ku-Mo**

### CODE REVIEW

File: src/web-blocks/divider.tsx [hunk 1]
```diff
@@ -1,19 +1,13 @@
 import * as React from "react";
 import { DividerHorizontalIcon } from "@radix-ui/react-icons";
-import { Styles } from "@chaibuilder/runtime/controls";
-import { ChaiBlock } from "../core/types/ChaiBlock.ts";
+import { ChaiBlockComponentProps, registerChaiBlockSchema, StylesProp, ChaiStyles } from "@chaibuilder/runtime";
+// import { ChaiBlock } from "../core/types/ChaiBlock.ts";
```
Comment (surajair) [inline src/web-blocks/divider.tsx]: Remove commented code

File: src/web-blocks/icon.tsx [hunk 2]
```diff
@@ -1,19 +1,20 @@
...
-  block: ChaiBlock & {
-    blockProps: Record<string, string>;
-    styles: Record<string, string>;
-  },
-) => {
-  const { blockProps, icon, styles, width, inBuilder, height } = block;
+export type IconBlockProps = {
+  styles: ChaiStyles;
+  icon: string;
+  width: string;
```
Comment (surajair) [inline src/web-blocks/icon.tsx]: width and height should be number

File: src/web-blocks/link.tsx [hunk 3]
```diff
@@ -68,14 +68,55 @@ const Config = {
...
+  ...registerChaiBlockSchema({
+    properties: {
+      styles: StylesProp(""),
+      content: {
+        type: "string",
+        default: "Link goes here",
+        title: "Content",
+      },
+      link: {
+        type: "string",
```
Comment (surajair) [inline src/web-blocks/link.tsx]: this should be an object and ui:widget as 'link'

**Review by surajair — CHANGES_REQUESTED** [review]

File: src/web-blocks/divider.tsx (hunk 1 above)
Comment (Rahul-ku-Mo) [reply src/web-blocks/divider.tsx]: OK will remove in next commit

File: src/web-blocks/icon.tsx (hunk 2 above)
Comment (Rahul-ku-Mo) [reply src/web-blocks/icon.tsx]: Noted

File: src/web-blocks/link.tsx (hunk 3 above)
Comment (Rahul-ku-Mo) [reply src/web-blocks/link.tsx]: Noted. I'll update this to object

File: src/web-blocks/image.tsx [hunk 4]
```diff
@@ -36,17 +35,66 @@ const Config = {
...
-    image: Image({
-      title: "Image",
-      default: "https://fakeimg.pl/400x200?text=Choose&font=bebas",
-    }),
-    alt: SingleLineText({ title: "Alt text", default: "", ai: true, i18n: true }),
-    width: SingleLineText({ title: "Width", default: "" }),
-    height: SingleLineText({ title: "Height", default: "" }),
-    lazyLoading: Checkbox({ title: "Lazy Load", default: true }),
-  },
+  // props: {
```
Comment (surajair) [inline src/web-blocks/image.tsx]: Removed dead code

File: src/web-blocks/image.tsx [hunk 5]
```diff
@@ -36,17 +35,66 @@ const Config = {
...
+        title: "Alt text",
+        default: "",
+        ui: {
+          "ui:placeholder": "Enter Here",
+        },
+      },
+      height: {
+        type: "number",
+        title: "Height",
+        default: null,
```
Comment (surajair) [inline src/web-blocks/image.tsx]: default must be `""`

File: src/web-blocks/image.tsx [hunk 6]
```diff
@@ -36,17 +35,66 @@ const Config = {
...
+  ...registerChaiBlockSchema({
+    properties: {
+      styles: StylesProp(""),
+      image: {
+        type: "string",
+        title: "Image",
+        default: "https://fakeimg.pl/400x200?text=Choose&font=bebas",
+        ui: {
+          "ui:widget": "image",
+          "ui:options": {
```
Comment (surajair) [inline src/web-blocks/image.tsx]: what does `ui:options` do?

**Review by surajair — CHANGES_REQUESTED** [review]

File: src/web-blocks/image.tsx (hunk 6 above)
Comment (Rahul-ku-Mo) [reply src/web-blocks/image.tsx]: I was trying something. So I used that, there is no such thing in the docs. It was autosuggested by cursor.

File: src/web-blocks/image.tsx (hunk 5 above)
Comment (Rahul-ku-Mo) [reply src/web-blocks/image.tsx]: Yeah, there was some type error when I gave it an empty value. Now its working fine after adding "". Will update in next commit .

**Review by surajair — APPROVED** [review]

**Pull request merged**